spring-cloud-deployer-local/src/main/java/org/springframework/cloud/deployer/spi/local/AbstractLocalDeployerSupport.java

```diff
@@ -85,6 +86,28 @@ public AbstractLocalDeployerSupport(LocalDeployerProperties properties) {
 		this.properties = properties;
 		this.javaCommandBuilder = new JavaCommandBuilder(properties);
 		this.dockerCommandBuilder = new DockerCommandBuilder();
+		this.restTemplate = buildRestTemplate(properties);
+	}
+
+	/**
+	 * Builds a {@link RestTemplate} used for calling app's shutdown endpoint. If
+	 * needed can be overridden from an implementing class. This default implementation
+	 * sets connection and read timeouts for {@link SimpleClientHttpRequestFactory} and
+	 * configures {@link RestTemplate} to use that factory. If shutdown timeout in
+	 * properties negative, returns default {@link RestTemplate} which doesn't use timeouts.
+	 *
+	 * @param properties the local deployer properties
+	 * @return the rest template
+	 */
+	protected RestTemplate buildRestTemplate(LocalDeployerProperties properties) {
+		if (properties != null && properties.getShutdownTimeout() > -1) {
```

Comment:
Is zero a valid timeout value?  Maybe we can add `@Min(0)` on the property itself?